presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -382,13 +392,51 @@ private void invokeNoMoreSplitsIfNecessary()
 
         // Bucketed partitions are fully loaded immediately since all files must be loaded to determine the file to bucket mapping
         if (tableBucketInfo.isPresent()) {
+            if (AcidUtils.isTransactionalTable(table.getParameters())) {
+                throw new PrestoException(NOT_SUPPORTED, format("Presto cannot read bucketed hive transactional tables: %s.%s", table.getDatabaseName(), table.getTableName()));
+            }
             return hiveSplitSource.addToQueue(getBucketedSplits(path, fs, splitFactory, tableBucketInfo.get(), bucketConversion));
         }
 
         // S3 Select pushdown works at the granularity of individual S3 objects,
         // therefore we must not split files when it is enabled.
         boolean splittable = getHeaderCount(schema) == 0 && getFooterCount(schema) == 0 && !s3SelectPushdownEnabled;
-        fileIterators.addLast(createInternalHiveSplitIterator(path, fs, splitFactory, splittable));
+        if (!AcidUtils.isTransactionalTable(table.getParameters())) {
+            fileIterators.addLast(createInternalHiveSplitIterator(path, fs, splitFactory, splittable));
+        }
+        else {
+            if (AcidUtils.isFullAcidTable(table.getParameters())) {
+                throw new PrestoException(NOT_SUPPORTED, format("Reading from Full Acid tables are not supported: %s.%s", table.getDatabaseName(), table.getTableName()));
+            }
+
+            // Now we should only have insert only table
+            if (!AcidUtils.isInsertOnlyTable(table.getParameters())) {
+                throw new PrestoException(HIVE_INVALID_METADATA, format("Unknown transactional table type [%s] : neither Insert Only nor Full Acid", table.getTableName()));
+            }
+            AcidUtils.Directory directory = AcidUtils.getAcidState(
+                    path,
+                    configuration,
+                    validWriteIds.orElseThrow(() -> new IllegalStateException("No validWriteIds present"))
+                            .getTableValidWriteIdList(table.getDatabaseName() + "." + table.getTableName()),
+                    false,
+                    true);
+
+            if (!directory.getOriginalFiles().isEmpty()) {
+                throw new PrestoException(
```

Comment:
1. These are files listing instead of directory locations like in case of delta and base directories.
2. The original files have same schema as base/delta files in case of Insert Only Tables
3. The original files do not have rowId fields, hence have different schema than Full ACID base/delta files
4. They need special handling during read path in Full ACID tables. This will be covered separately and we are almost done with those changes (will be a new PR for those).

For Insert Only Tables, we can create splits directly out of these files instead of iterating over their location. The work being done for (4) covers that and hence I continued to keep them unsupported for now.